File: frontend/types/alerts.ts
export interface Alert {
  id: number;
  ticker: string;
  alert_type: string;
  severity: string;
  message: string;
  dismissed: boolean;
  created_at: string;
  previous_value: number | null;
  current_value: number | null;
}

export interface AlertsResponse {
  alerts: Alert[];
  count: number;
  undismissed_count: number;
}

File: backend/app/schemas/alerts.py
from pydantic import BaseModel


class AlertResponse(BaseModel):
    id: int
    ticker: str
    alert_type: str
    severity: str
    message: str
    dismissed: bool = False
    created_at: str = ""
    previous_value: float | None = None
    current_value: float | None = None

    model_config = {"from_attributes": True}


class AlertsListResponse(BaseModel):
    alerts: list[AlertResponse]
    count: int
    undismissed_count: int
